components/brave_wallet/brave_wallet_utils.cc

```diff
@@ -133,4 +137,48 @@ std::string Uint256ValueToHex(uint256_t input) {
   return out;
 }
 
+std::string GenerateMnemonic(const std::vector<uint8_t>& entropy) {
+  uint8_t* entropy_in = const_cast<uint8_t*>(entropy.data());
+  size_t size_in = entropy.size();
+
+  // If input entropy is empty, we generate a random 128 bits
+  std::vector<uint8_t> bytes(16);
+  if (entropy.empty()) {
+    crypto::RandBytes(&bytes[0], bytes.size());
+    entropy_in = bytes.data();
+    size_in = bytes.size();
+  }
+
+  // entropy size should be 128-256 bits
```

Comment:
This should check that the value is not only in the range of 128-256 but also adheres to the correct sizing I listed above, which I will list again for brevity: 128, 160, 192, 224, 256.

Reply:
fixed in ca4d806d885cb83113b7163a9861ffdbfde28203